pkg/agent/billing/billing.go

```diff
@@ -287,6 +371,26 @@ func (s *metricsState) drainEnqueue(logger *zap.Logger, conf *Config, hostname s
 			StopTime:       now,
 			Value:          int(math.Round(history.total.activeTime.Seconds())),
 		})))
+		countInBatch += 1
+		queue.enqueue(logAddedEvent(logger, billing.Enrich(now, hostname, countInBatch, batchSize, &billing.IncrementalEvent{
+			MetricName:     conf.IngressBytesMetricName,
+			Type:           "", // set by billing.Enrich
+			IdempotencyKey: "", // set by billing.Enrich
+			EndpointID:     key.endpointID,
+			StartTime:      s.pushWindowStart,
+			StopTime:       now,
+			Value:          int(history.totalIngressBytes),
+		})))
+		countInBatch += 1
+		queue.enqueue(logAddedEvent(logger, billing.Enrich(now, hostname, countInBatch, batchSize, &billing.IncrementalEvent{
+			MetricName:     conf.EgressBytesMetricName,
+			Type:           "", // set by billing.Enrich
+			IdempotencyKey: "", // set by billing.Enrich
+			EndpointID:     key.endpointID,
+			StartTime:      s.pushWindowStart,
+			StopTime:       now,
+			Value:          int(history.totalEgressBytes),
+		})))
```

Comment:
Another piece of disabling code if it's not used: Can we avoid emitting events if the VM has network usage tracking disabled? (I think "are the metrics zero" might be an ok-enough proxy). Reasoning is that it allows us to use dummy values for the `MetricName` in the config, meaning that we can deploy without waiting on billing team.

nb: I guess it requires more complex around `batchSize` :)

Reply:
Added - just had to loop over the history twice to compute the batch size.